### PR TITLE
[Task]: Add parameter types getGetter/getSetterCode

### PR DESCRIPTION
**Pull request by mcop1**
Add types for all parameters, including $language parameter
Removed setter functions for `CalculatedValue`, since they are not used, change return values for getPhpdocInputType/getParameterTypeDeclaration accordingly

### CODE REVIEW

**Comment by jdreesen** [comment]
The title is wrong, it should be *parameter type* ;)

**Comment by mcop1** [comment]
> The title is wrong, it should be _parameter type_ ;)

Thanks

File: models/DataObject/ClassDefinition/Data/CalculatedValue.php [hunk 1]
```diff
@@ -510,7 +510,7 @@ public function getSetterCodeLocalizedfields($class)
         $code .= '* @param ' . $this->getPhpdocInputType() . ' $' . $key . "\n";
         $code .= '* @return $this' . "\n";
         $code .= '*/' . "\n";
-        $code .= 'public function set' . ucfirst($key) . '(' . '$' . $key . ', $language = null): static' . "\n";
+        $code .= 'public function set' . ucfirst($key) . '(' . '$' . $key . ', ?string $language = null): static' . "\n";
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/CalculatedValue.php]: Should we add also the missing parameter type here?
```suggestion
        $code .= 'public function set' . ucfirst($key) . '(' . $this->getParameterTypeDeclaration() . ' $' . $key . ', ?string $language = null): static' . "\n";
```

But this is missing also in the other getSetterCode methods...

Comment (mcop1) [reply models/DataObject/ClassDefinition/Data/CalculatedValue.php]: I was thinking about doing another pr, but I guess it is ok to do it in the same pr, so thanks for the suggestion!

**Comment by blankse** [comment]
Why do empty setters exist for CalculatedValue? Should we remove them completely?

**Comment by blankse** [comment]
The methods getParameterTypeDeclaration()/getPhpdocInputType() must return string "mixed"

**Comment by mcop1** [comment]
> The methods getParameterTypeDeclaration()/getPhpdocInputType() must return string "mixed"

I changed the `getSetterCodeXXX` functions to return an empty string and adapted the testcases. I don´t think there is a point in keeping the setters, what do you think?

**Comment by blankse** [comment]
I also think that we do not need them.

**Pull request merged**